react_app/src/components/FlagsContainer.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+import { makeStyles } from '@material-ui/core/styles';
+
+import Grid from '@material-ui/core/grid'
+import Paper from '@material-ui/core/paper'
+import CTFCategory from './CTFCategory'
+
+import { Query } from 'react-apollo'
+import gql from 'graphql-tag'
+
+const GET_ALL_CATEGORIES_QUERY = gql`
+    {
+      allCategories{
+        id
+        title
+        order
+      }
+    }
+`;
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    padding: "1em",
+  },
+}));
+
+function getCategories(){
+    return(
+        <Query query={GET_ALL_CATEGORIES_QUERY} pollInterval={60000}>{
+            ({ loading, error, data }) => {
+              if (loading) return <div>Fetching</div>
+              if (error) return <div>Error</div>
+
+              const stuffToRender = data.allCategories
+              return (
+                  stuffToRender.map(category => <CTFCategory key={category.id} category={category} />)
```

Comment:
Eew variable name :( how about this refactor that uses restructuring to fish out the property from `data` you are interested in at the point you get it;

```
 ({ loading, error, { allCategories }) => {
              if (loading) return <div>Fetching</div>
              if (error) return <div>Error</div>

              return (
                  allCategories.map(category => <CTFCategory key={category.id} category={category} />)
              )
```

Reply:
This is neater, although named variables are generally considered OK outside of JS...

This is a much neater way of expressing this functionality, although the first line seems to be doing some weird syntactic hack which I'm still attempting to work out.

Reply:
It's not the presence of a named variable, it's just the name `stuffToRender`